### PR TITLE
WRN-9786: Migrated unit tests for i18nDecorator to testing-library

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated unit tests for i18nDecorator to testing-library

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-9786

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: packages/i18n/I18nDecorator/tests/I18nDecorator-specs.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import {shallow, mount} from 'enzyme';
 import ilib from 'ilib';
+import '@testing-library/jest-dom';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import {updateLocale} from '../../locale';
 import {I18nContextDecorator, I18nDecorator} from '../I18nDecorator';
```
Comment (adrian-cocoara-lgp) [inline packages/i18n/I18nDecorator/tests/I18nDecorator-specs.js:7]: sort imports alphabetically

File: packages/i18n/I18nDecorator/tests/useI18n-specs.js [hunk 2]
```diff
@@ -1,14 +1,13 @@
-import {shallow} from 'enzyme';
 import {useState} from 'react';
+import '@testing-library/jest-dom';
+import {render, screen} from '@testing-library/react';
 
 import {updateLocale} from '../../locale';
 
 import useI18n from '../useI18n';
```
Comment (adrian-cocoara-lgp) [inline packages/i18n/I18nDecorator/tests/useI18n-specs.js:7]: sort imports alphabetically

File: packages/i18n/I18nDecorator/tests/useI18n-specs.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
-import {shallow} from 'enzyme';
 import {useState} from 'react';
+import '@testing-library/jest-dom';
+import {render, screen} from '@testing-library/react';
 
 import {updateLocale} from '../../locale';
```
Comment (adrian-cocoara-lgp) [inline packages/i18n/I18nDecorator/tests/useI18n-specs.js]: delete this empty line

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Review by alexandrumorariu — APPROVED** [review]
LGTM

**Pull request merged**